cloudflare/resource_cloudflare_access_group_test.go

```diff
@@ -0,0 +1,359 @@
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudflareAccessGroup_Updated(t *testing.T) {
+	var before, after cloudflare.AccessGroup
+	name := "cloudflare_access_group.test"
+	resourceName := strings.Split(name, ".")[1]
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareAccessGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareAccessGroupConfigBasic(resourceName, accountID, email),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareAccessGroupExists("cloudflare_access_group.test", &before),
+				),
+			},
+			{
+				Config: testAccCheckCloudflareAccessGroupConfigNewValue(name, accountID, email),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareAccessGroupExists("cloudflare_access_group.test", &after),
+					testAccCheckCloudflareAccessGroupIDUnchanged(&before, &after),
+					resource.TestCheckResourceAttr(name, "include.email", "test-changed@example.com"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudflareAccessGroup_CreateAfterManualDestroy(t *testing.T) {
+	var before, after cloudflare.AccessGroup
+	var initialID string
+	name := "cloudflare_access_group.test"
+	resourceName := strings.Split(name, ".")[1]
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareAccessGroupDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudflareAccessGroupConfigBasic(resourceName, accountID, email),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareAccessGroupExists("cloudflare_access_group.test", &before),
+					testAccManuallyDeleteAccessGroup("cloudflare_access_group.test", &initialID),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+			{
+				Config: testAccCheckCloudflareAccessGroupConfigNewValue(name, accountID, email),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareAccessGroupExists("cloudflare_access_group.test", &after),
+					testAccCheckCloudflareAccessGroupRecreated(&before, &after),
+					resource.TestCheckResourceAttr(
+						"cloudflare_access_group.test", "account_id", accountID),
+					resource.TestCheckResourceAttr(
+						"cloudflare_access_group.test", "name", fmt.Sprintf("%s/updated", name)),
+					resource.TestCheckResourceAttr(
+						"cloudflare_access_group.test", "include.email", email),
+					resource.TestCheckResourceAttr(
+						"cloudflare_access_group.test", "exclude.email", email),
+					resource.TestCheckResourceAttr(
+						"cloudflare_access_group.test", "require.email", email),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudflareAccessGroup_UpdatingAccountIDForcesNewResource(t *testing.T) {
+	var before, after cloudflare.AccessGroup
+	name := "cloudflare_access_group.test"
+	resourceName := strings.Split(name, ".")[1]
+	newAccountID := "01a7362d577a6c3019a474fd6f485634"
```

Comment:
We don't currently have another test zone but perhaps if this is needed for a real resource, we can create it in CI.